modules/core/misc/python/package/mat_wrapper/__init__.py

```diff
@@ -4,15 +4,15 @@
 import cv2 as cv
 from typing import TYPE_CHECKING, Any
 
-# Type subscription is not possible in python 3.8
+# Same as cv2.typing.NumPyArrayGeneric, but avoids circular dependencies
 if TYPE_CHECKING:
-    _NDArray = np.ndarray[Any, np.dtype[np.generic]]
+    _NumPyArrayGeneric = np.ndarray[Any, np.dtype[np.generic]]
```

Comment:
I could actually import cv2.typing here so that the types don't fall out of sync. But the alias would still be necessary to avoid circular dependencies.